### DISCUSSION TITLE
deepFreeze

### DISCUSSION THREAD

**Discussion by nkitsaini** [remeda/remeda#895] [Suggest a Function]
The built-in `Object.freeze` is not recursive, it would be better to have a recursive implementation.

**Comment by eranhirsch** [comment]
When using typescript, when do you find you need to use freeze vs just marking an object as readonly? type-fest offers a deep readonly utility. Freeze is problematic as it ignores mutations silently instead of preventing them from happening.

**Reply by nkitsaini** [reply]
So I maintain everything is immutable, but references can be shared across multiple objects in one of my project having lots of objects.

With `TypeFest` I can be sure that the current piece of code doesn't mutate the object but I still have to worry about rest of the code to not mutate the object but with `deepFreeze`, I know that if I return a "freezed" object no one will be able to modify it. 

It is like having type is good, but it would be even better to have the guarantee provided by runtime itself.


> Freeze is problematic as it ignores mutations silently instead of preventing them from happening.

That is only in the non-strict mode right?

**Reply by eranhirsch** [reply]
I agree. Let's do it! Can you send a PR for this?

**Reply by nkitsaini** [reply]
Yeah, sure. Will send a PR
